src/Core/src/Platform/iOS/LabelExtensions.cs

```diff
@@ -88,9 +88,16 @@ internal static void UpdateTextHtml(this UILabel platformLabel, ILabel label)
 			};
 
 			NSError nsError = new();
-#pragma warning disable CS8601
-			platformLabel.AttributedText = new NSAttributedString(text, attr, ref nsError);
-#pragma warning restore CS8601
+			CoreFoundation.DispatchQueue.MainQueue.DispatchAsync(() =>
```

Comment:
I might not be following your description quite right but should we check if we're on a background thread before dispatching?

Reply:
With something like this:

```c#
NSError nsError = new();
if (NSThread.IsMain)
{
	platformLabel.AttributedText = new NSAttributedString(text, attr, ref nsError);
	if (label.Handler is ILabelHandler labelHandler)
	{
		labelHandler.UpdateValue(nameof(ILabel.TextColor));
		labelHandler.UpdateValue(nameof(ILabel.Font));
	}
	label.InvalidateMeasure();
}
else
{
	CoreFoundation.DispatchQueue.MainQueue.DispatchAsync(() =>
	{
		platformLabel.AttributedText = new NSAttributedString(text, attr, ref nsError);
		if (label.Handler is ILabelHandler labelHandler)
		{
			labelHandler.UpdateValue(nameof(ILabel.TextColor));
			labelHandler.UpdateValue(nameof(ILabel.Font));
		}
		label.InvalidateMeasure();
	});
}
```

The list view/collection view with the HTML label looks right, but the app crashes when using the collection view
<img width="1454" alt="Screenshot 2024-12-14 at 00 59 02" src="https://github.com/user-attachments/assets/e4b46448-45fc-4df6-810d-6abc1771282f" />